Support for slots to handles crypto API changes under SPM

itayzafrir opened the pull request:
Support for the API changes from slots to handles under SPM (code separation between SPE and NSPE). 
Same idea as in https://github.com/ARMmbed/mbed-crypto/blob/development/library/psa_crypto.c#L29-L42

Patater commented on library/psa_crypto_slot_management.c:
```diff
@@ -26,6 +26,20 @@
+ * In case MBEDTLS_PSA_CRYPTO_SPM is defined the code is built for SPM (Secure
+ * Partition Manager) integration which separate the code into two parts
+ * NSPE (Non-Secure Process Environment) and SPE (Secure Process Environment).
+ * In this mode an additional header file should be included.
```
Could we change this to the following?

```
When MBEDTLS_PSA_CRYPTO_SPM is defined, the code is being build for SPM (Secure Partition Manager) integration which separates the code into two parts: NSPE (Non-Secure Processing Element) and SPE (Secure Processing Element). When building for the SPE, an additional header file should be included.
```

itayzafrir replied:
Changed

Patater replied:
I see this still says "Environment" instead of "Element".

gilles-peskine-arm replied:
I don't see where “environment” should be changed to “element”. (N)SPE is (Non-)Secure Processing Environment.

Patater commented on library/psa_crypto_slot_management.c:
```diff
@@ -26,6 +26,20 @@
+ */
+#if defined(MBEDTLS_PSA_CRYPTO_SPM)
+/*
+ * PSA_CRYPTO_SECURE means that this file is compiled to the SPE side.
```
```diff
- compiled to the SPE side.
- compiled for the SPE.
```

itayzafrir replied:
Changed

Patater commented on library/psa_crypto_slot_management.c:
```diff
@@ -26,6 +26,20 @@
+#if defined(MBEDTLS_PSA_CRYPTO_SPM)
+/*
+ * PSA_CRYPTO_SECURE means that this file is compiled to the SPE side.
+ * some headers will be affected by this flag.
```
```diff
- some headers will be affected by this flag.
+ Some headers will be affected by this flag.
```

itayzafrir replied:
Changed

Patater commented:
If you wouldn't mind, it'd help to have another commit to update the other comment block. Thanks.

itayzafrir commented:
@Patater  I've updated the the code comments, commit message title and description, I also added a new commit to update existing SPM documentation in the code. 
Please take another look.
